engine/equations.cc

```diff
@@ -774,7 +774,27 @@ namespace MathDAG
           }
       }
     return r;
-  }     
+  }
+  
+  VariableDAGPtr SystemOfEquations::getNodeFromVar(const VariableBase& v)
+  {
+    NodePtr r;
+    if (expressionCache.exists(v))
+      return dynamic_pointer_cast<VariableDAG>(expressionCache[v]);
+    else if (v.type()!=VariableBase::undefined) r=makeDAG(const_cast<VariableBase&>(v));
+    return dynamic_pointer_cast<VariableDAG>(r);
+  }         
+
+  ostringstream SystemOfEquations::getDefFromIntVar(const VariableBase& v)
+  {
+    ostringstream o;
+    
+	std::vector<VariableDAG*>::const_iterator it1=std::find_if(integrationVariables.begin(),integrationVariables.end(),[&](VariableDAG* i){return i->valueId==v.valueId();});         
+    VariableDAGPtr input=expressionCache.getIntegralInput((*it1)->valueId);    
```

Comment:
No need for the find_if on line 792. We already have valueId as v.valueId()

Reply:
Removed, thanks.